ERCS/erc-7894.md

```diff
@@ -0,0 +1,143 @@
+   - **PointUpdated**: Triggered when token points are adjusted.
+
+7. **Voting Struct**
+   - **votingIndex**: uint256
+   - **name**: string
+   - **description**: string
+   - **tokenIndex**: uint256
+   - **point**: uint256
+   - **increase**: bool
+   - **yes**: uint256
+   - **no**: uint256
+   - **startDate**: uint256
+   - **endDate**: uint256
+   - **isActive**: bool
+
+## Rationale
+
+SocialSBT decouples governance power from token holdings by implementing a community-based governance system. This ensures fairness and prevents undue influence by high-point token holders. 
+
+Unlike traditional DAOs, where governance power is directly tied to the number of tokens held, SocialSBT enforces a **contribution-based governance model**. However, social points are **not automatically assigned** when a user contributes. Instead, members must perform valuable actions that prompt the community to initiate a voting process to recognize their efforts. This approach ensures that social points are **collectively decided and transparently adjusted** rather than arbitrarily awarded.  
+
+Here’s how SocialSBT mitigates whale attacks in practice:
+
+1. **Non-transferable tokens**: Since SocialSBT tokens cannot be bought or transferred, governance power cannot be acquired through market purchases.
+2. **Community-driven social point adjustments**: Social points are not automatically granted. Instead, a proposal must be initiated and approved by the DAO community for a member’s points to increase or decrease.
+3. **Contribution-based voting power**: Only members who provide meaningful contributions can gain governance influence, and these contributions must be recognized by the community through an **on-chain voting process**.
+
+Because SocialSBT ensures that governance influence is directly linked to community engagement rather than wealth, whales cannot manipulate voting outcomes by simply acquiring tokens. This system reinforces democratic participation and aligns decision-making with the collective interests of the DAO.
+
+## Backwards Compatibility
+
+SocialSBT extends the [ERC-721](./eip-721) standard while remaining compatible with its core functionalities. However, it introduces non-transferability and a social point mechanism, which deviate from the standard.
+
+## Test Cases
+
+1. **Minting Tokens**
+   - Expected: Successful minting for eligible users.
+   - Failures:
+     - User already owns a token.
+     - Payment amount does not match _price.
+
+2. **Voting**
+   - Expected: Correct vote recording and prevention of duplicate voting.
+   - Failures:
+     - Non-token holder participation.
+     - Duplicate votes on the same proposal.
+
+3. **Ending Voting**
+   - Expected: Accurate point adjustment based on majority votes.
+   - Failures:
+     - Ending before expiration.
+     - Inactive proposals.
+
+4. **Burning Tokens**
+   - Expected: Permanent token removal.
+   - Failures:
+     - Unauthorized burn attempts.
```

Comment:
Test cases should be either simple inputs and outputs (i.e. function calls and state changes) or actual coded unit tests.